include/grpc++/impl/codegen/status.h

```diff
@@ -59,6 +69,8 @@ class Status {
   /// Return the (binary) error details.
   // Usually it contains a serialized google.rpc.Status proto.
   grpc::string error_details() const { return binary_error_details_; }
+  /// Return the full fidelity error string, which includes all child errors.
```

Comment:
Note this is an API change (with the new ctor). I think if you do not want to defer this PR, it would be better to (for now) move the two into private and add a friend function/class to get and set it and mark the new function/class experimental. 

The name error_string is not very clear (mainly because we already have error_message and error_details...), maybe it should be called debug_error_string? Does it only get set when there is a non-OK status?

The error_string only applies to the client side, right? Maybe the new API can be on ClientContext instead...

Reply:
Good call. I will move this to be an experimental API for ClientContext.

This does only get set when the error is != GRPC_ERROR_NONE